SingularityUI/app/components/globalSearch/GlobalSearch.jsx

```diff
@@ -0,0 +1,115 @@
+let globalSearchClasses;
+import React from 'react';
+import classNames from 'classnames';
+
+import { Typeahead } from 'react-typeahead';
+import fuzzy from 'fuzzy';
+
+class GlobalSearch extends React.Component {
+  constructor(...args) {
+    super(...args);
+    this.optionSelected = this.optionSelected.bind(this);
+    this.resetSelection = this.resetSelection.bind(this);
+    this.clear = this.clear.bind(this);
+    this.focus = this.focus.bind(this);
+    this.componentDidUpdate = this.componentDidUpdate.bind(this);
+    this.render = this.render.bind(this);
+  }
+
+  optionSelected(requestIdObject) {
+    let requestId = this.getValueFromOption(requestIdObject);
+    app.router.navigate(`/request/${ requestId }`, { trigger: true });
+    this.clear();
+    return this.props.onHide();
+  }
+
+  resetSelection() {
+    return this.refs.typeahead.setState({
+      selectionIndex: 0
+    });
+  }
+
+  clear() {
+    this.refs.typeahead.setEntryText('');
+    return this.resetSelection();
+  }
+
+  focus() {
+    this.refs.typeahead.focus();
+    return this.resetSelection();
+  }
+
+  searchOptions(inputValue, options) {
+    // fuzzy lazily just appends a string before and after a matching char
+    // we have to later use a simple shift-in shift-out state machine to convert
+    let fuzzyOptions = {
+      returnMatchInfo: true
+    };
+
+    let searched = fuzzy.filter(inputValue, options, fuzzyOptions);
+
+    return searched;
+  }
+
+  renderOption(option, index) {
+    // transform fuzzy string into react component
+    let bolded = option.string.map(function(matchInfo) {
+      if (matchInfo.match) {
+        return <b>{matchInfo.char}</b>;
+      } else {
+        return <span>{matchInfo.char}</span>;
+      }
+    });
+
+    return <span>{bolded}</span>;
+  }
+
+  getValueFromOption(option) {
+    return option.original;
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.visible && (this.props.visible !== prevProps.visible)) {
+      return this.focus();
+    }
+  }
+
+  render() {
+    let options = _.pluck(this.props.requests.toJSON(), 'id');
+
+    globalSearchClasses = classNames({
```

Comment:
add `const globalSearchClasses` here instead of the `let` at the top of the file